src/plugins/console/public/lib/osd/osd.ts

```diff
@@ -36,69 +38,102 @@ import {
   ListComponent,
   TemplateAutocompleteComponent,
   UsernameAutocompleteComponent,
+  SharedComponent,
 } from '../autocomplete/components';
 
-import $ from 'jquery';
-import _ from 'lodash';
-
-import Api from './api';
+import { Api } from './api';
+
+export type IdAutocompleteComponentFactory = (
+  name: string,
+  parent: SharedComponent,
+  multiValued?: boolean
+) => IdAutocompleteComponent;
+
+export type ComponentFactory = (
+  name: string,
+  parent: SharedComponent | null,
+  multiValued?: boolean | unknown
+) => SharedComponent;
+
+export interface ParametrizedComponentFactories {
```

Comment:
I see this `ParametrizedComponentFactories` type is used in multiple places. Should we create a common folder or some type file to host it?

Reply:
1. You right, we should move it, I will add it to types file that I created in my other PR. I forgot to mention the order of merging, it should first be https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4008, then https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4148 and only after that this one.

Reply:
I see. Then this PR is blocked by #4008 and #4148. Let's get back to this PR after them merged. @curq I will start reviewing these two.

Reply:
I moved it to `types.ts` files that holds the common types